src/haddock/gear/zerofill.py

```diff
@@ -0,0 +1,110 @@
+"""
+Handles zero filling prefix for module folder names.
+
+To facilitate folder sorting, HADDOCK3 module run folder names have a
+integer prefix starting at 0.
+
+Depending on the number of modules of the run, the integer prefix is a
+single digit number or multi-digit number (00, 01, 02, 03...).
+
+This gear module contains the "zero filling" logic in HADDOCK 3.
+
+The main class of this module `zerofill` is already instantiated such
+that its state can be used and edited throughout the HADDOCK3 library.
+
+Examples
+--------
+>>> from haddock.gear import zerofill
+>>> zerofill.read(modules)  # a dictionary containing 11 modules
+>>> zerofill.zfnum
+2
+>>> zerofill.fill("topoaa", 0)
+"00_topoaa"
+"""
+from math import ceil, log10
+
+
+class _ZeroFill:
+    """
+    Register the zero fill number.
+
+    Examples
+    --------
+    >>> _ZeroFill()
+    >>> zerofill.read(modules)  # a dictionary containing five modules
+    >>> zerofill.zfnum
+    1
+    >>> zerofill.fill("topoaa", 0)
+    "0_topoaa"
+    """
+
+    def __init__(self, zfnum=None):
+        self.zfnum = zfnum
+        return
+
+    def read(self, modules):
+        """
+        Register the zerofill number for current run.
+
+        Zero fill number if available through the attribute: `zfnum`.
+
+        Parameters
+        ----------
+        iterable
+            Usually a dictionary or a list of the modules names.
+            Any object implementing `len`.
+        """
+        self.zfnum = get_zerofill_for_modules(modules)
```

Comment:
The called function is one line, is there a reason the define it externally? If there is none I would vote for the definition to be written inline.